src/test/java/org/example/test.java

```diff
@@ -0,0 +1,74 @@
+package org.example;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import org.junit.jupiter.api.Test;
+
+public class test {
+
+    private final int n = 10;
+    private final double a = -1;
+    private final double b = 1;
+    private final double delta = 0.01;
+
+    private final double[] x = new double[n + 1];
```

Comment:
Ne vem, kakšna je praksa v Javi, ampak sam bi se izogibal globalnim spremenljivkam, ki jih spreminjate s funkcijami. Tabele `x` in `weights` bi zapakiral v razred in jih izračunal z metodami znotraj razreda.